app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class Users::SessionsController < Devise::SessionsController
+  after_action :assign_guest_cart_to_user,only: [:create]
+  after_action :after_logout_setup, only: [:destroy]
+
+  # before_action :configure_sign_in_params, only: [:create] :
+
+  # GET /resource/sign_in
+  # def new
+  #   super
+  # end
+
+  # POST /resource/sign_in
+  def create
+
+    super
+  end
+
+  # DELETE /resource/sign_out
+  def destroy
+    super
+
+  end
+
+  def assign_guest_cart_to_user
+
+    session[:guest]=false
+    @shopping_cart=Cart.find_by(id:session[:shopping_cart]['id'])
+    @shopping_cart.update!(user_id:current_user.id)
+    User.find_by(id:session[:guest_user_id]).destroy
+    session[:guest_user_id]=nil
+    session[:shopping_cart]=@shopping_cart
+    
```

Comment:
Please review this function.

Reply:
This function is working as expected. is there any logical error or syntax error?